pkg/reconciler/vertex/controller.go

```diff
@@ -434,3 +437,22 @@ func (r *vertexReconciler) markPhaseLogEvent(vertex *dfv1.Vertex, log *zap.Sugar
 	vertex.Status.MarkPhaseFailed(reason, message)
 	r.recorder.Event(vertex, corev1.EventTypeWarning, reason, message)
 }
+
+func checkChildrenResourceStatus(ctx context.Context, c client.Client, vertex *dfv1.Vertex) error {
+	// fetch the pods for calculating the status of child resources
+	var podList corev1.PodList
+	if err := c.List(ctx, &podList,
+		client.InNamespace(vertex.GetNamespace()),
+		client.MatchingLabels{dfv1.KeyAppName: vertex.GetName()},
```

Comment:
LabelSelector should be `dfv1.KeyPipelineName=pipelineName,dfv1.KeyVertexName=vertex.Spec.Name`